docs/main/user-guides/local-storage-user-guide/local-pv-hostpath/advanced-operations/xfs-quota/enable-xfs-quota.md

```diff
@@ -0,0 +1,166 @@
+
+## Install the OpenEBS Dynamic LocalPV Provisioner
+
+Refer to the [OpenEBS Installation documentation](../../../../../quickstart-guide/installation.md) to install the OpenEBS LocalPV Hostpath Provisioner.
+
+## Create StorageClass
+
+1. To create a hostpath StorageClass with the XFSQuota configuration option, use the following YAML definition. This configuration will enable the XFSQuota for the specified path and storage type.
+
+```
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: openebs-hostpath-xfs
+  annotations:
+    openebs.io/cas-type: local
+    cas.openebs.io/config: |
+      - name: StorageType
+        value: "hostpath"
+      - name: BasePath
+        value: "/var/openebs/local/"
+      - name: XFSQuota
+        enabled: "true"
+provisioner: openebs.io/local
+volumeBindingMode: WaitForFirstConsumer
+reclaimPolicy: Delete
+```
+
+2. For advanced configuration of XFSQuota, you may also set the `softLimitGrace` and `hardLimitGrace` parameters, which define the storage capacity limits beyond the Persistent Volume (PV) storage request. The updated YAML definition is as follows:
+
+```
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: openebs-hostpath-xfs
+  annotations:
+    openebs.io/cas-type: local
+    cas.openebs.io/config: |
+      - name: StorageType
+        value: "hostpath"
+      - name: BasePath
+        value: "/var/openebs/local/"
+      - name: XFSQuota
+        enabled: "true"
+        data:
+          softLimitGrace: "0%"
+          hardLimitGrace: "0%"
+provisioner: openebs.io/local
+volumeBindingMode: WaitForFirstConsumer
+reclaimPolicy: Delete
+```
+
+:::note
+- `softLimitGrace` and `hardLimitGrace` are used in conjunction with the PV storage request to determine the soft and hard limits of the quota.
+
+- The size of these limits is calculated as **"Size of PV storage request * (1 + LimitGrace%)"**
+
+- If no values are specified, the default is **softLimitGrace: "0%" / hardLimitGrace: "0%"**, meaning the storage capacity is limited to the PV storage request value.
+  
+  For example, with a PV of 100Gi capacity and values **softLimitGrace: "90%" / hardLimitGrace: "100%"**, the soft limit will be set to 190Gi, and the hard limit will be set to 200Gi.
+  You can select to use either `softLimitGrace` or `hardLimitGrace` independently based on your requirements.
+
+  Refer to the relevant [xfs_quota documentation](https://man7.org/linux/man-pages/man8/xfs_quota.8.html#QUOTA_OVERVIEW) for more detailed instructions regarding the configuration of soft and hard limits.
+:::
+
+## Create a PVC
+
+1. To create a PVC using the StorageClass's name, use the following definition:
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: local-hostpath-xfs
+spec:
+  storageClassName: openebs-hostpath-xfs
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 5Gi
```

Comment:
Don't see fsType specified as xfs in the pvc. Is this expected?
cc @niladrih

Reply:
The filesystem type is not configurable. A device with a filesystem is a prerequisite. The hostpath provisioner doesn't require the configurable filesystem type parameter that we see with the CSI drivers (on the storageClass though). The CSI drivers carve out new logical devices from the lvm-vg/zfs-pool/spdk-blob, and hence they ask for configuration related to what filesystem they should use.

There are checks in the vol creation stage which restricts XFS quota specific operations to filesystems with XFS characteristics (linux df based magic number check).